courses/views_test.py

```diff
@@ -62,6 +108,79 @@ def courses():
     return CourseFactory.create_batch(3)
 
 
+@pytest.fixture
```

Comment:
This shouldn't be a fixture the way it's currently written because it's just a plain function.

If you want to go with a fixture, you'd want to set it up with [indirect parameterization](https://docs.pytest.org/en/7.3.x/example/parametrize.html#indirect-parametrization) with `num_progams` and `num_courses` fixtures that use `request.param` or a default so that you'd be able configure it like this:

```python
@pytest.mark.parametrize("num_progams", [100], indirect=True)
@pytest.mark.parametrize("num_courses", [3], indirect=True)
def test_func(course_catalog): ...
```

Reply:
Yes, fair. Thank you! On it.

Reply:
I opted instead to just parameterize the values a la https://github.com/mitodl/mitxonline/pull/1962/files#r1373812611 - it really is a function, so I also renamed `course_catalog_api` to `populate_course_catalog_data`.